### PR TITLE
chore: add flag to enable/disable deprecated APIs

### PR DESCRIPTION
**Pull request by darrenvechain**
# Description

- Disable deprecated endpoints by default
- Add a flag to enable them

PR here to fix E2E: https://github.com/vechain/thor-e2e-tests/pull/58

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

**Review by leszek-vechain — DISMISSED** [review]

File: cmd/thor/flags.go [hunk 1]
```diff
@@ -69,6 +69,10 @@ var (
 		Value: 1000,
 		Usage: "limit the number of logs returned by /logs API",
 	}
+	apiEnabledDeprecatedFlag = cli.BoolFlag{
+		Name:  "api-enabled-deprecated",
+		Usage: "enable deprecated APIs",
```
Comment (libotony) [inline cmd/thor/flags.go:74]: Can you include the endpoints in the description?

File: api/subscriptions/subscriptions.go [hunk 2]
```diff
@@ -50,10 +51,11 @@ const (
 	pingPeriod = (pongWait * 7) / 10
 )
 
-func New(repo *chain.Repository, allowedOrigins []string, backtraceLimit uint32, txpool *txpool.TxPool) *Subscriptions {
+func New(repo *chain.Repository, allowedOrigins []string, backtraceLimit uint32, txpool *txpool.TxPool, deprecatedEnabled bool) *Subscriptions {
```
Comment (libotony) [inline api/subscriptions/subscriptions.go:54]:
```suggestion
func New(repo *chain.Repository, allowedOrigins []string, backtraceLimit uint32, txpool *txpool.TxPool, enableDeprecated bool) *Subscriptions {
```

It's better to have the same name convention.

Comment (darrenvechain) [reply api/subscriptions/subscriptions.go]: Pushed commit

**Review by libotony — DISMISSED** [review]

File: cmd/thor/main.go [hunk 3]
```diff
@@ -80,6 +79,7 @@ func main() {
 			apiCallGasLimitFlag,
 			apiBacktraceLimitFlag,
 			apiAllowCustomTracerFlag,
+			apiEnableDeprecatedFlag,
```
Comment (leszek-vechain) [inline cmd/thor/main.go:82]: wonder if we should keep some name convention here:

- apiCallGasLimitFlag
- apiBacktraceLimitFlag
- apiAllowCustomTracerFlag
- apiEnableDeprecatedFlag
- enableAPILogsFlag
- apiLogsLimitFlag

so should we also change enableAPILogsFlag to apiEnableLogsFlag ?

**Review by libotony — APPROVED** [review]

File: cmd/thor/main.go (hunk 3 above)
Comment (otherview) [reply cmd/thor/main.go:82]: Makes sense, but I wouldn't block this PR for that change. Plus, naming has a way of opening up can of worms 😅

**Pull request merged**